core/trino-main/src/main/java/io/trino/connector/CatalogPruneTask.java

```diff
@@ -127,14 +131,22 @@ public ThreadPoolExecutorMBean getExecutor()
     }
 
     @VisibleForTesting
-    void pruneWorkerCatalogs()
+    public void pruneWorkerCatalogs()
     {
         Set<ServiceDescriptor> online = selector.selectAllServices().stream()
                 .filter(descriptor -> !nodeInfo.getNodeId().equals(descriptor.getNodeId()))
                 .collect(toImmutableSet());
 
         // send message to workers to trigger prune
         List<CatalogHandle> activeCatalogs = getActiveCatalogs();
+        executePruneOnServices(online, activeCatalogs);
+
+        // prune all inactive catalogs - we pass an empty set here because manager always retains active catalogs
+        connectorServicesProvider.pruneCatalogs(ImmutableSet.of());
+    }
+
+    void executePruneOnServices(Set<ServiceDescriptor> online, List<CatalogHandle> activeCatalogs)
```

Comment:
This is also pruning on workers "services" in the names below is just an artifact of the discovery system.  I'd just name this `pruneWorkerCatalogs` also